docs/configuration.md

```diff
@@ -1 +1,363 @@
+
+  ```
+  ENV_VAR1 = 1
+  ENV_VAR3 = 3
+  ENV_VAR4 = 4
+  MY_VAR = my-value
+  ```
+  
+  #### Dynamic Plugins
+  
+  The Operator can configure [Dynamic Plugins](https://github.com/janus-idp/backstage-showcase/blob/main/showcase-docs/dynamic-plugins.md). To support Dynamic Plugins, the Backstage deployment should contain a dedicated initContainer called **install-dynamic-plugins** (see [RHDH deployment.yaml](../config/manager/default-config/deployment.yaml)). To enable the Operator to configure Dynamic Plugins for a specific Backstage instance (CR), the user must create a ConfigMap with an entry called **dynamic-plugins.yaml**.
+  
+  For example, the **dynamic-plugins-config** ConfigMap contains a simple Dynamic Plugins configuration, which includes predefined default plugins in **dynamic-plugins.default.yaml** and the GitHub plugin provided in the package located at `./dynamic-plugins/dist/backstage-plugin-catalog-backend-module-github-dynamic`.
+  
+  ```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: dynamic-plugins-config
+data:
+  dynamic-plugins.yaml: |
+    includes:
+      - dynamic-plugins.default.yaml
+    plugins:
+      - package: './dynamic-plugins/dist/backstage-plugin-catalog-backend-module-github-dynamic'
+        disabled: false
+        pluginConfig:
+          catalog:
+            providers:
+              github:
+                organization: "${GITHUB_ORG}"
+                schedule:
+                  frequency: { minutes: 1 }
+                  timeout: { minutes: 1 }
+                  initialDelay: { seconds: 100 }
+  ```
+
+To configure it with the Backstage CR, the following spec should be included:
+
+  ```yaml
+spec:
+  application:
+    dynamicPluginsConfigMapName: "dynamic-plugins-config"
+  ```
+
+#### Deployment Parameters
+
+  **NOTE:** These fields are deprecated for versions **>= 0.3.0** in favor of the [spec.deployment](#deployment-configuration).
+  
+  ```yaml
+spec:
+  application:
+    image: Backstage container image: [tag], e.g., "quay.io/my/my-rhdh:latest"
+    replicas: number of replicas, e.g., 2
+    imagePullSecrets: array of image pull secrets names, e.g.,
+      - my-secret-name
+  ```
+
+#### Route
+
+To support Backstage service routing on OpenShift, the Operator can create a `route.openshift.io` resource, as specified in the **spec.application.route** field. Here’s an example:
+
+  ```yaml
+spec:
+  application:
+    route:
+      enabled: true
+      host: my.host
+      subdomain: mysubdomain
+      tls:
+        certificate: "certificatecontent"
+        externalCertificateSecretName: "my-certificate"
+        key: "keycontent"
+        caCertificate: "caCertificatecontent"
+  ```
+  
+  Here, the user can specify some of the OpenShift Route specifications fields. The names of the Backstage `spec.application.route` fields correspond to the names of Route specifications fields and follow the same default rules if not specified.
+  
```

Comment:
I'd suggest calling this out as well. See https://github.com/redhat-developer/rhdh-operator/blob/main/api/v1alpha2/backstage_types.go#L278-L279

```suggestion
  
  Also note that securing Routes with external certificates in TLS secrets (via the `spec.application.route.tls.externalCertificateSecretName` CR field) is a Technology Preview feature in OpenShift. It requires enabling the `RouteExternalCertificate` OpenShift Feature Gate and might not be functionally complete. See [Creating a route with externally managed certificate](https://docs.openshift.com/container-platform/4.16/networking/routes/secured-routes.html#nw-ingress-route-secret-load-external-cert_secured-routes) for more details.

```

Reply:
This was marked as resolved, but I couldn't see these lines. Does it make sense to call it out here?

Reply:
fixed